ceph/rados/core_workflows.py

```diff
@@ -1033,3 +1033,82 @@ def get_journalctl_log(
         except Exception as er:
             log.error(f"Exception hit while command execution. {er}")
         return log_lines
+
+    def get_osd_list(self):
+        """
+        Used to get the list of OSDs configured in the cluster nodes
+        Returns:
+            Dictionary with the keys as cluster osd names and values are the list of osd names configured to the node.
+        """
+        try:
+            osd_list = dict()
+            cmd = "ceph osd  tree"
+            osds = self.run_ceph_command(cmd)
+            for entry in osds["nodes"]:
+                if entry["type"] == "host":
+                    osd_list[entry["name"]] = entry["children"]
+            return osd_list
+        except Exception as e:
+            log.error("Failed to  get the osd list : %s", str(e))
+
+    def get_osd_pid(self, host_object, osd):
```

Comment:
I would suggest to create generic method to get any Ceph daemon PID. 
This will be helpful to perform any system activity on node.

Reply:
We could move this generic methods to generic directory/module(example. , utils.py).

Reply:
Hope this would be helpful, https://github.com/ceph/teuthology/blob/main/teuthology/orchestra/daemon/systemd.py#L102-L129